lib/decidim/cdtb/spam/user_spam_detector.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require "csv"
+
+module Decidim
+  module Cdtb
+    module Spam
+      # Detect spam behavior in users
+      #
+      class UserSpamDetector < ::Decidim::Cdtb::Task
+        include ActiveSupport::Configurable
+
+        URL_REGEX = %r{(http|ftp|https)://([\w_-]+(?:(?:\.[\w_-]+)+))([\w.,@?^=%&:/~+#-]*[\w@?^=%&/~+#-])}.freeze
+
+        config_accessor :spam_words, :spam_regexp
+
+        def initialize(organization = nil)
+          @organization = organization
+          progress_bar = { title: "Decidim::User" }
+          super("SPAM DETECTOR", progress_bar: progress_bar)
+        end
+
+        def prepare_execution(_ctx)
+          @users = if @organization.present?
+                     Decidim::User.where(organization: @organization)
+                   else
+                     Decidim::User.all
+                   end
+
+          @num_users = @users.count
+          log_task_info("Checking #{@num_users} users...")
+        end
+
+        def total_items
+          @num_users
+        end
+
+        def do_execution(context)
+          progress_bar = context[:progress_bar]
+
+          CSV.open("spam_users.csv", "w") do |csv|
+            csv_headers = ["ID", "Is suspicious?", "Name", "Email", "Nickname", "Personal URL", "About"]
+            csv << csv_headers
+
+            @users.find_each do |user|
+              suspicious = "❌"
+
+              if spam_user?(user)
+                suspicious = "✅"
+                @num_applied+= 1
+              end
```

Comment:
Avoid exporting emojis, better nil/x or TRUE/FALSE or yes/not i don't know the standard way of exporting a boolean
```suggestion
              suspicious = false

              if spam_user?(user)
                suspicious = true
                @num_applied+= 1
              end
```